### PR TITLE
docs(data-provider): update data provider doc

### PR DESCRIPTION
**Pull request by alicanerdurmaz**
added : http://refine.dev/docs/api-reference/core/providers/data-provider/

### CODE REVIEW

File: documentation/docs/api-reference/core/providers/data-provider.md [hunk 1]
```diff
@@ -4,99 +4,25 @@ title: Data Provider
...
-
-:::
+<div>
+    <img src="https://refine.ams3.cdn.digitaloceanspaces.com/website/static/img/guides-and-concepts/providers/data-provider/tutorial_dataprovider_flog.png" />
+</div>
 
 :::note
-Data hooks use [React Query](https://react-query.tanstack.com/) to manage data fetching. React Query handles important concerns like caching, invalidation, loading states, etc.
-
+Data hooks use [TanStack Query](https://tanstack.com/query) to manage data fetching. React Query handles important concerns like caching, invalidation, loading states, etc.
```
Comment (salihozdemir) [inline documentation/docs/api-reference/core/providers/data-provider.md:18]:
```suggestion
Data hooks use [TanStack Query](https://tanstack.com/query) to manage data fetching. It handles important concerns like caching, invalidation, loading states, etc.
```

File: documentation/docs/api-reference/core/providers/data-provider.md [hunk 2]
```diff
@@ -276,934 +153,551 @@ const PostList: React.FC = () => {
...
-const { mutate } = useCreate();
+-   **Using `dataProviderName` prop in the data hooks **and all **data-related** components/**functions.**
 
-mutate({
-    resource: "categories",
-    values: {
-        title: "New Category",
-    },
+```tsx title="posts/list.tsx"
+useTable<IPost>({
```
Comment (salihozdemir) [inline documentation/docs/api-reference/core/providers/data-provider.md:182]:
```suggestion
```tsx
useTable({
```

File: documentation/docs/api-reference/core/providers/data-provider.md [hunk 3]
```diff
@@ -276,934 +153,551 @@ const PostList: React.FC = () => {
...
+                    name: "products",
+                    options: {
+                        // Refine will use the `exampleDataProvider` data provider for this resource
+                        dataProviderName: "exampleDataProvider",
+                    },
+                },
+            ]}
+        />
+    );
+};
```
Comment (salihozdemir) [inline documentation/docs/api-reference/core/providers/data-provider.md:214]:
```suggestion
```tsx
        <Refine
            dataProvider={{
                default: defaultDataProvider,
                example: exampleDataProvider,
            }}
            resources={[
                {
                    // Refine will use the `default` data provider for this resource
                    name: "posts",
                },
                {
                    name: "products",
                    options: {
                        // Refine will use the `exampleDataProvider` data provider for this resource
                        dataProviderName: "exampleDataProvider",
                    },
                },
            ]}
        />
```

File: documentation/docs/api-reference/core/providers/data-provider.md [hunk 4]
```diff
@@ -276,934 +153,551 @@ const PostList: React.FC = () => {
...
 
-> `TVariables` is a user defined type which can be passed to [`useCreateMany`](/docs/api-reference/core/hooks/data/useCreateMany/) to type `variables`
-
-<br/>
-
-**refine** will consume this `createMany` method using the `useCreateMany` data hook.
-
-```ts
-import { useCreateMany } from "@pankod/refine-core";
+:::caution
```
Comment (salihozdemir) [inline documentation/docs/api-reference/core/providers/data-provider.md:219]: I think, this caution no to be in here. Maybe we don't need it. We can describe it without caution.

**Review by salihozdemir — APPROVED** [review]

File: documentation/docs/api-reference/core/providers/data-provider.md [hunk 5]
```diff
@@ -276,934 +153,552 @@ const PostList: React.FC = () => {
...
+const App = () => {
+    return (
+        <Refine
+            dataProvider={{
+                default: defaultDataProvider,
+                example: exampleDataProvider,
+            }}
+            resources={[
+                {
+                    // Refine will use the `default` data provider for this resource
```
Comment (omeraplak) [inline documentation/docs/api-reference/core/providers/data-provider.md:201]:
```suggestion
                    // refine will use the `default` data provider for this resource
```

File: documentation/docs/api-reference/core/providers/data-provider.md [hunk 6]
```diff
@@ -276,934 +153,552 @@ const PostList: React.FC = () => {
...
+                    name: "posts",
+                },
+                {
+                    name: "products",
+                    options: {
+                        // Refine will use the `exampleDataProvider` data provider for this resource
```
Comment (omeraplak) [inline documentation/docs/api-reference/core/providers/data-provider.md:207]:
```suggestion
                        // refine will use the `exampleDataProvider` data provider for this resource
```

**Review by omeraplak — CHANGES_REQUESTED** [review]

File: documentation/docs/api-reference/core/providers/data-provider.md [hunk 7]
```diff
@@ -276,934 +153,634 @@ const PostList: React.FC = () => {
...
-:::note
-    `getMany`, `createMany`, `updateMany` and `deleteMany` properties are optional. If you don't implement them, Refine will use `getOne`, `create`, `update` and `deleteOne` methods to handle multiple requests. If your API supports these methods, you can implement them to improve performance.
-:::
-
-<br/>
-
-### `create`
-
-This method allows us to create a single item in a resource.
+:::info
```
Comment (alicanerdurmaz) [inline documentation/docs/api-reference/core/providers/data-provider.md:257]:
```suggestion
:::info-tip
```

File: documentation/docs/api-reference/core/providers/data-provider.md [hunk 8]
```diff
@@ -276,934 +153,634 @@ const PostList: React.FC = () => {
...
+```
 
-:::note
-Fake REST API is based on [JSON Server Project](https://github.com/typicode/json-server). **Simple REST Dataprovider** is fully compatible with the REST rules and methods of the **JSON Server**.
 :::
 
-<br />
+You can pick data providers in two ways:
+
+-   **Using `dataProviderName` prop in the data hooks **and all **data-related** components/**functions.**
```
Comment (aliemir) [inline documentation/docs/api-reference/core/providers/data-provider.md:179]:
```suggestion
-   **Using `dataProviderName` prop in the data hooks and all data-related components/functions.**
```

**Review by aliemir — APPROVED** [review]

**Pull request merged**